miasm2/analysis/depgraph.py

```diff
@@ -417,27 +462,27 @@ def _resolve_depNode(self, depnode):
             # Intra-block resolving
             ## Get dependencies
             read = set()
+
             modifier = False
 
             for affect in self._get_affblock(depnode):
                 if affect.dst == depnode.element:
-                    ### Avoid tracking useless elements, as XOR EAX, EAX
-                    src = expr_simp(affect.src)
-
-                    read.update(src.get_r(mem_read=True, cst_read=True))
+                    elements = self.follow_apply_cb(affect.src)
+                    read.update(elements)
                     modifier = True
 
             ## If it's not a modifier affblock, reinject current element
             if not modifier:
-                read = set([depnode.element])
+                read = set([(True, depnode.element)])
```

Comment:
To avoid confusion, you could use a `NamedTuple` here, for instance with `follow` and `element` attribute.
The code snippets which are using this structure will be less prone to error and more intelligible.